src/shims/fs.rs

```diff
@@ -722,6 +799,248 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         this.try_unwrap_io_result(result)
     }
+
+    fn mkdir(
+        &mut self,
+        path_op: OpTy<'tcx, Tag>,
+        mode_op: OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        this.check_no_isolation("mkdir")?;
+
+        let _mode = if this.tcx.sess.target.target.target_os.to_lowercase() == "macos" {
```

Comment:
```suggestion
        let _mode = if this.tcx.sess.target.target.target_os.as_str() == "macos" {
```
I got rid of the `to_lowercase` everywhere else, as it turns out we do not need them.